corehq/apps/userreports/views.py

```diff
@@ -138,6 +139,7 @@ def create_report(request, domain):
 
 class ReportBuilderView(BaseDomainView):
 
+    @method_decorator(require_permission(Permissions.edit_data))
     @cls_to_view_login_and_domain
```

Comment:
Sorry for not asking more clearly. I was wondering if `cls_to_view_login_and_domain` was also doing a permission check at a lower level (which would be redundant and unnecessary), but looking at it again I think it's probably fine.